asset_dashboard/forms.py

```diff
@@ -1,12 +1,32 @@
-from django.forms import ModelForm, CharField
-from django import forms
-from .models import Project, Section
+from django.forms import ModelForm, CharField, TextInput, Textarea
+from .models import Project, Section, ProjectScore
 
 class ProjectForm(ModelForm):
     class Meta:
         model = Project
-        fields = ['name', 'description', 'section_owner']
+        fields = ['name', 'description', 'section_owner', 'category']
         widgets = {
-            'name': forms.TextInput(attrs={'class': 'form-control'}),
-            'description': forms.Textarea(attrs={'class': 'form-control'})
+            'name': TextInput(attrs={'class': 'form-control w-50'}),
+            'description': Textarea(attrs={'class': 'form-control'})
         }
+
+
+class ProjectScoreForm(ModelForm):
+    class Meta:
+        model = ProjectScore
+        fields = [
+            'core_mission_score',
+            'operations_impact_score',
+            'sustainability_score',
+            'ease_score',
+            'geographic_distance_score',
+            'social_equity_score'
+        ]
+
+    def __init__(self, *args, **kwargs):
+        super(ProjectScoreForm, self).__init__(*args, **kwargs)
+
+        for field_name, field in self.fields.items():
+            field.widget.attrs['min'] = ProjectScore.min_value
```

Comment:
i think you should be able to get this from the field object?

Reply:
Not sure I understand. Are any of these answers near what you're talking about? https://stackoverflow.com/questions/3647805/get-models-fields-in-django

Reply:
Any idea where to get these values from the field object? This is what is returned from a field [within the form class's](https://github.com/datamade/ccfp-asset-dashboard/blob/44d696ec196881bfa63a91fafb4fea606d81d9f6/asset_dashboard/forms.py#L29) `__init__`:
```
{'max_value': None, 'min_value': None, 'required': True, 'label': 'Social equity score', 'initial': 1, 'show_hidden_initial': False, 'help_text': '', 'disabled': False, 'label_suffix': None, 'localize': False, 'widget': <django.forms.widgets.NumberInput object at 0x7ff615486dc0>, 'error_messages': {'required': 'This field is required.', 'invalid': 'Enter a whole number.'}, 'validators': []}
```

Reply:
i'm surprised that `validators` is empty!

Reply:
yeah it's a head scratcher

Reply:
@fgregg Here is a commit with a different approach: 455308f. I'm not a big fan of that verbose `ProjectScore.__dict__[field_name].field.validators[0].limit_value` for a number of reasons:
1. what if the order of the array changes and `validators[0]` doesn't match `attrs['min']`? Is there a way to ensure that doesn't happen?
2. the naming isn't very descriptive
3. it's ugly

You mentioned a form field for the model. Is this what you were talking about? https://docs.djangoproject.com/en/dev/howto/custom-model-fields/#specifying-the-form-field-for-a-model-field

I think I see how that might work. If I created these form field for the ProjectScore form and model, shouldn't I also refactor the other form classes and widgets into a model form field? For the sake of consistency.

Reply:
i cleaned this up a bit with https://github.com/datamade/ccfp-asset-dashboard/pull/20/commits/ffae69de054b0d5b59f5c87347cce03af794400b